catkit/interfaces/Instrument.py

```diff
@@ -167,9 +241,20 @@ def _forced_safe_close(self):
     def __close(self):
         # __func() can't be overridden without also overriding those that call it.
         try:
+            if self.run_as_service:
+                self.stop_service()
+
             if self.instrument:
-                self._close()
-                self.log.info("Safely closed connection to '{}'".format(self.config_id))
+                try:
+                    self._close()
+                    self.log.info("Safely closed connection to '{}'".format(self.config_id))
+                except Exception:
+                    if not self.run_as_service:
+                        raise
```

Comment:
I'll have to remember why I added this... 🤔

Reply:
Right, the re-open below. Originally, ``service_thread`` was daemonic, but I no longer think that's a good idea.